qiita_pet/handlers/download.py

```diff
@@ -393,3 +394,38 @@ def get(self):
 
                         self._set_nginx_headers(zip_fn)
         self.finish()
+
+
+class DownloadPublicArtifactHandler(BaseHandlerDownload):
+    @coroutine
+    @execute_as_transaction
+    def get(self):
+        artifact_id = self.get_argument("artifact_id", None)
+
+        if artifact_id is None:
+            raise HTTPError(422, reason='You need to specify an artifact id')
+        else:
+            try:
+                artifact = Artifact(artifact_id)
+            except QiitaDBUnknownIDError:
+                raise HTTPError(422, reason='Artifact does not exist')
+            else:
+                if artifact.visibility != 'public':
+                    raise HTTPError(422, reason='Artifact is not public. If '
```

Comment:
Should this be a 404 to avoid leaking information about non public data?

Reply:
I don't have a strong preference but thought that after the last 2 releases where we are making things more public it will make more sense to add these kind of errors/warnings so if another user gets it and it is a mistake it can reported to us. A good example is that a user publishes one of these ids but forgets to make public.

Reply:
But they would report the issue with a 404 too right? We [respond](https://github.com/biocore/qiita/blob/master/qiita_pet/handlers/study_handlers/base.py#L30) with a 404 when a user attempts to access a study description that they don't have access too.